src/system.transactions/tests/CloneTxTests.cs

```diff
@@ -0,0 +1,124 @@
+        [InlineData(CloneType.RollbackDependent, IsolationLevel.Snapshot, true, true, TransactionStatus.Committed)]
+        [InlineData(CloneType.RollbackDependent, IsolationLevel.Chaos, true, true, TransactionStatus.Committed)]
+        [InlineData(CloneType.RollbackDependent, IsolationLevel.Unspecified, true, true, TransactionStatus.Committed)]
+        */
+        public void Run(CloneType cloneType, IsolationLevel isoLevel, bool forcePromote, bool completeClone, TransactionStatus expectedStatus )
+        {
+            TransactionOptions options = new TransactionOptions
+            {
+                IsolationLevel = isoLevel,
+                // Shorten the delay before a timeout for blocking clones.
+                Timeout = TimeSpan.FromSeconds(1)
+            };
+
+            CommittableTransaction tx = new CommittableTransaction(options);
+
+            Transaction clone;
+            switch (cloneType)
+            {
+                case CloneType.Normal:
+                    {
+                        clone = tx.Clone();
+                        break;
+                    }
+                case CloneType.BlockingDependent:
+                    {
+                        clone = tx.DependentClone(DependentCloneOption.BlockCommitUntilComplete);
+                        break;
+                    }
+                case CloneType.RollbackDependent:
+                    {
+                        clone = tx.DependentClone(DependentCloneOption.RollbackIfNotComplete);
+                        break;
+                    }
+                default:
+                    {
+                        throw new Exception("Unexpected CloneType - " + cloneType.ToString());
+                    }
+            }
+
+            if (forcePromote)
+            {
+                HelperFunctions.PromoteTx(tx);
+            }
+
+            Assert.True(clone.IsolationLevel == tx.IsolationLevel, "Clone's IsolationLevel does not match original");
+            Assert.True(clone.TransactionInformation.Status == tx.TransactionInformation.Status, "Clone's Status does not match original");
+            Assert.True(clone.TransactionInformation.LocalIdentifier == tx.TransactionInformation.LocalIdentifier, "Clone's LocalIdentifier does not match original");
+            Assert.True(clone.TransactionInformation.DistributedIdentifier == tx.TransactionInformation.DistributedIdentifier, "Clone's DistributedIdentifier does not match original");
+
+            CommittableTransaction cloneCommittable = clone as CommittableTransaction;
+            Assert.True(cloneCommittable == null, "Clone is castable as a CommittableTransaction");
+
+            try
+            {
+                tx.Commit();
+            }
+            catch (TransactionAbortedException)
+            { }
+
+            Assert.True(tx.TransactionInformation.Status == expectedStatus, "Unexpected TransactionStatus (" + tx.TransactionInformation.Status.ToString());
```

Comment:
This could be:
```C#
Assert.Equal(expectedStatus, tx.TransactionInformation.Status);
```
Then if the assert fails, the resulting exception message will include both the expected and the actual status.

Same goes for some of the Assert.True's earlier.